Complex inte test

antoine-de opened the pull request:
test with mock chaos disruption

antoine-de committed:
Tests: add cli argument to mock kraken

antoine-de committed:
Jormun: add test with mock chaos

kinnou02 commented on source/jormungandr/jormungandr/interfaces/v1/fields.py:
```diff
@@ -367,7 +367,7 @@ def output(self, key, obj):
     "status": fields.String(),
     "updated_at": DateTime(),
-    "tags": NonNullList(NonNullNested(fields.String())),
+    "tags": fields.String(),
```
a list seem a good thing

antoine-de commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,167 @@
+        self._exchange = Exchange('navitia', durable=True, delivry_mode=2, type='topic')
+        self.mock_chaos_connection.connect()
+
+    def teardown(self):
```
need to close connection no ?

antoine-de committed:
Disruption: corrections after review

l-vincent-l commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,222 @@
+@dataset([("main_routing_test", ['--BROKER.rt_topics=bob', 'spawn_maintenance_worker'])])
+class TestChaosDisruptions(ChaosDisruptionsFixture):
+    """
+    Note: it is done as a new fixture, the spawn a new kraken, in order not the get previous disruptions
```
it's not english

l-vincent-l commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,222 @@
+        self.send_chaos_disruption("bob_the_disruption", "stopB", "stop_area")
+
+        #we sleep a bit to let kraken reload the data
+        sleep(0.3)
```
It'll be safer to call kraken until there's new data

l-vincent-l commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,222 @@
+        """
+        when calling the pt object stopB, at first we have no disruptions,
+
+        then we mock a disruption sent from chaos, and we call again the pt object B
```
stopB

l-vincent-l commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,222 @@
+        #at first we got only one disruption on A
+        assert len(disruptions) == 1
+
+        assert 'bob_the_disruption' in impacts_by_uri
```
might be clearer to use an any

l-vincent-l commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,222 @@
+@dataset([("main_routing_test", ['--BROKER.rt_topics=bob', 'spawn_maintenance_worker'])])
+class TestChaosDisruptions2(ChaosDisruptionsFixture):
+    """
+    Note: it is done as a new fixture, the spawn a new kraken, in order not the get previous disruptions
```
same as above

l-vincent-l commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,222 @@
+
+            impacts_by_uri = {d['uri']: d for d in disruptions}
+
+            #at first we got only one disruption on A
```
B

l-vincent-l commented on source/jormungandr/tests/chaos_disruptions_tests.py:
```diff
@@ -0,0 +1,222 @@
+
+    disruption = feed_entity.Extensions[chaos_pb2.disruption]
+
+    #disruption = chaos_pb2.Disruption()
```
You might want to delete it, unless it's a comment

l-vincent-l commented on source/jormungandr/tests/disruptions_tests.py:
```diff
@@ -103,9 +103,7 @@ def test_disruption_end_point(self):
-        simple disruption call
-
-        without param we don't get the stop areas, so we'll get the impact on the line and on the network
+        when calling the pt object stopA, we should get it's disruptions
```
its

antoine-de committed:
Disruption: add a last_rt_data_loaded field

to be able to now the date of the last reload of data due to rt, add a
field with the datetime of the last rt data integration.

the field is not serialized (I don't see the point)
